drivers/flash/flash_stm32l4x.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2017 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include <flash_stm32.h>
+
+#define STM32L4X_BANK_SIZE_MAX	512
+#define STM32L4X_PAGE_SHIFT	11
+
+#define STM32L4X_FLASH_END \
+	((u32_t)(STM32L4X_BANK_SIZE_MAX << STM32L4X_PAGE_SHIFT) - 1)
+
+/* offset and len must be aligned on 8, positive and not beyond end of flash */
+bool flash_stm32_valid_range(off_t offset, u32_t len)
+{
+	return offset % 8 == 0 &&
+	       len % 8 == 0 &&
+	       offset >= 0 &&
+	       (offset + len - 1 <= STM32L4X_FLASH_END);
+}
+
+/* STM32L4xx devices can have up to 512 2K pages on two 256x2K pages banks */
+static unsigned int flash_stm32l4x_get_page(off_t offset)
+{
+	return offset >> STM32L4X_PAGE_SHIFT;
+}
+
+int flash_stm32_check_status(struct flash_stm32_priv *p)
```

Comment:
Right, I need to check for F3, but it's OK.

The only tweak is to add a:
#if defined(FLASH_FLAG_PGPERR)
along the:
#if defined(FLASH_FLAG_RDERR)